### ISSUE TITLE
[feature request] autocomplete in form field

### ISSUE THREAD

**Issue by lovasoa** [lovasoa/SQLpage#221]
### Discussed in https://github.com/lovasoa/SQLpage/discussions/220

<div type='discussions-op-text'>

<sup>Originally posted by **heejit** February 10, 2024</sup>
First SQLPage  is very nice many thanks for sharing.

How do I allow user to select / auto complete a input box from sql table which contain 10K rows (item_names) user need to input few letter to start search.

</div>

**Comment by lovasoa** [comment]
What I would do is reuse the "tags input" element from the tabler css library that SQLPage embeds:

![image](https://github.com/lovasoa/SQLpage/assets/552629/924af4d6-7188-4ffa-b4ac-e420def88c8e)

```html
<div class="mb-3">
  <label class="form-label">Tags input</label>
  <select type="text" class="form-select tomselected ts-hidden-accessible" placeholder="Select tags" id="select-tags" value="" multiple="multiple" tabindex="-1">
    <option value="HTML">HTML</option>
    <option value="JavaScript">JavaScript</option>
    <option value="CSS">CSS</option>
    <option value="jQuery">jQuery</option>
    <option value="Bootstrap">Bootstrap</option>
    <option value="Ruby">Ruby</option>
    <option value="Python">Python</option>
  </select><div class="ts-wrapper form-select multi"><div class="ts-control"><input tabindex="0" role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-controls="select-tags-ts-dropdown" id="select-tags-ts-control" placeholder="Select tags" type="select-multiple"></div></div>
</div>
```

https://tabler.io/preview

You can [follow the instructions to create your own custom component](https://sql.ophir.dev/custom_components.sql), or contribute directly to the [form component](https://github.com/lovasoa/SQLpage/blob/main/sqlpage/templates/form.handlebars)

The way I would do it is adding a new row level parameter called `options_source` that the user would set to the name of a sql file to call to get the list of options.

Then the user could do 
### `fom.sql`
```sql
select 'form' as component;
select 'My choice' as name, 'my_items.sql' as options_source;
```
### `my_items.sql`
```sql
SELECT 'json' AS component, (
    SELECT JSON_GROUP_ARRAY(item_name) FROM items WHERE item_name LIKE '%' || $search
) AS contents;
```

what do you think ?

**Comment by heejit** [comment]
We can use input-box with list tag 

<input id="id1" list="a-datalist">

<datalist id="a-datalist">
<option value="item_1"> </option>
</datalist>

**Comment by lovasoa** [comment]
This is also a possibility, but the rendering is a little bit less pretty:

![image](https://github.com/lovasoa/SQLpage/assets/552629/0ba6fcdb-9825-4f7d-950e-04dd202b4d12)

**Comment by lovasoa** [comment]
Support for autocomplete has been added and will be released in 0.20.3 :tada: 

You will be able to use a `select` input form with a `dropdown` property and create a beautiful input with autocomplete. 

https://sql.ophir.dev/documentation.sql?component=form#component
